integration/restValidation.ts

```diff
@@ -0,0 +1,215 @@
+        // eslint-disable-next-line camelcase,@typescript-eslint/camelcase
+        index: Number(validator_index),
+        slot: Number(slot),
+        pubKey: pubkey,
+    }));
+    return proposers.filter(
+        ({slot, index}: Proposer) => validatorIndex === index && (slot > ignoreBefore || ignoreBefore === undefined),
+    );
+};
+
+const isBlockProposedFactory = (baseUrl: string) => async (slot: number): Promise<boolean> => {
+    const url = new URL(`/eth/v1/beacon/blocks/${slot}`, baseUrl);
+    try {
+        const response = await axios.get(url.href);
+        if (Number(response.data.data.message.slot) === slot) {
+            return true;
+        }
+    } catch {
+        return false;
+    }
+    return false;
+};
+
+export const restValidation = ({
+    baseUrl,
+    getValidatorPrivateKey,
+    limit,
+}: {
+    baseUrl: string;
+    getValidatorPrivateKey: () => Promise<SecretKey>;
+    limit: number;
+}): Promise<{
+    proposer: {
+        proposed: number;
+        delegated: number;
+    };
+    attestation: {
+        attestations: number;
+        delegated: number;
+    };
+}> =>
+    new Promise((resolve) => {
+        (async (): Promise<void> => {
+            await initBLS("herumi");
+            const validatorPrivateKey = await getValidatorPrivateKey();
+            const validatorPublicKey = validatorPrivateKey.toPublicKey().toHex();
+            console.log("Starting validator " + validatorPublicKey);
+
+            const logger = new WinstonLogger({module: "ChainGuardian", level: LogLevel.verbose});
+            const slashingProtection = sinon.createStubInstance(CGSlashingProtection);
+
+            const eth2API = new CgEth2ApiClient(config, baseUrl);
+            const validatorService = new Validator({
+                slashingProtection,
+                api: eth2API,
+                config,
+                secretKeys: [validatorPrivateKey],
+                logger,
+                graffiti: "ChainGuardian",
+            });
+
+            let firstSlot: number | undefined;
+            let slot: number | undefined;
+
+            let startEpoch = 1;
+            let lastEpoch = 1;
+
+            const getCommittees = getCommitteesFactory(baseUrl);
+            let committees: Committees[] = [];
+            const attestations = new Map<number, boolean>();
+
+            const getProposer = getProposerFactory(baseUrl);
+            const isBlockProposed = isBlockProposedFactory(baseUrl);
+            const proposers = new Map<number, boolean>();
+
+            const validatorUrl = new URL(`/eth/v1/beacon/states/head/validators/${validatorPublicKey}`, baseUrl);
+            const validator = await axios.get(validatorUrl.href);
+            const validatorIndex = Number(validator.data.data.index);
+
+            const url = new URL(`/eth/v1/events?topics=attestation,head`, baseUrl);
+            const eventSource = new EventSource(url.href);
```

Comment:
you already have eth2API instance why not use that to listen on events?

Reply:
probably is this problem? https://github.com/ChainSafe/ssz/issues/23
I tried validator on `Teku` Beacon node and got some error while `szz` deserializing

Reply:
I think we should extend CgEth2Api with specific implementation for different clients until they all converge to be fully compliant with spec.

Like `CgTekuEth2Api extends CgEth2Api` and `CgLighthouseEth2Api extends CgEth2Api` where CgEth2Api is spec compliant implementation

Reply:
> probably is this problem? ChainSafe/ssz#23
I tried validator on Teku Beacon node and got some error while szz deserializing

Nope, you shouldn't get empty beacon blocks over API. Send me json you get from Teku, there is probably some discrepancies with data we get.

Reply:
> I think we should extend CgEth2Api with specific implementation for different clients until they all converge to be fully compliant with spec.
> 
> Like `CgTekuEth2Api extends CgEth2Api` and `CgLighthouseEth2Api extends CgEth2Api` where CgEth2Api is spec compliant implementation

and this will cause the problem mentioned above https://github.com/NodeFactoryIo/ChainGuardian/pull/541#discussion_r597473300
then will be better to provide `client` and `version` to the client and if there is some non-compliant we can override for specific (or range) `client`/`version`

Reply:
But you know what client you will use for each testnet so you can provide that info in this PR and instantiate appropriate beacon api client instance. As for chainguardian, ideas was (and should still be) to query https://ethereum.github.io/eth2.0-APIs/#/Node/getNodeVersion and depending on that initialize either LighthouseBeaconApi instance or TekuBeaconApi instance or throw error if not supported.

Reply:
it is `out of scope` for this task
for Teku there is #529 where we will proceed by this https://github.com/NodeFactoryIo/ChainGuardian/pull/541#discussion_r597486236

Reply:
It makes no sense to merge this PR with all those new beacon api implementation. If this Pr is only for Lighthouse testing, than just use CgEth2BeaconApi everywhere and we can have separate PR with teku specific implementation.

Reply:
then what we gonna do with #529  ?

Reply:
Ok then, just use CgEth2BeaconApi here (in this PR) and add other api client implementations as part of #529 

Log spec discrepancies so we can report it upstream.